test/smoke/testApps/SimpleTraceData/src/main/java/com/microsoft/ajl/simplecalc/SimpleCalculatorServlet.java

```diff
@@ -0,0 +1,120 @@
+package com.microsoft.ajl.simplecalc;
+
+import com.microsoft.ajl.simplecalc.model.BinaryCalculation;
+import com.microsoft.ajl.simplecalc.model.BinaryOperator;
+import com.microsoft.applicationinsights.TelemetryClient;
+import com.microsoft.applicationinsights.telemetry.SeverityLevel;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.microsoft.ajl.simplecalc.ParameterConstants.*;
+
+/**
+ * Servlet implementation class SimpleCalculatorServlet
+ */
+@WebServlet(description = "Performs given calculation", urlPatterns = { "/doCalc" })
+public class SimpleCalculatorServlet extends HttpServlet {
+	private static final long serialVersionUID = -633683109556605395L;
+
+	/**
+	 * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse response)
+	 */
+	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		BinaryCalculation bc = null;
+		try {
+			bc = readParameters(request.getParameterMap());
+		}
+		catch (CalculatorParameterException cpe) {
+			String errMsg = cpe.getLocalizedMessage();
+			System.err.println(errMsg);
+			response.sendError(HttpServletResponse.SC_BAD_REQUEST, errMsg);
+			return;			
+		}
+		
+		if (bc == null) {
+			System.out.println("No parameters given.");
+			response.setStatus(HttpServletResponse.SC_NO_CONTENT);
+			return;
+		}
+		
+		response.setContentType("text/html;charset=UTF-8");
+		renderHtml(bc, response.getWriter());
+		
+		new TelemetryClient().trackTrace("This is first trace message.");
```

Comment:
Instead of creating a new instance of TelemetryClient for every trackTrace() call, I would suggest to create a Class variable for TelemtryClient and use it for sending multiple trackTrace() calls. It is advisable to minimize the creating of new instance of telemetry client instances as much as possible.

Reply:
@dhaval24 Thanks for your advice, I will do it.